lib/node_modules/@stdlib/string/base/distances/hamming-grapheme-clusters/lib/main.js

```diff
@@ -0,0 +1,87 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var format = require( '@stdlib/string/format' );
+var len = require('@stdlib/string/num-grapheme-clusters');
+var nextGraphemeClusterBreak = require( '@stdlib/string/next-grapheme-cluster-break' );
+
+
+// MAIN //
+
+/**
+* Calculates the Hamming distance between two strings with equal number of grapheme clusters.
+*
+* ## Notes
+*
+* -   The function returns a sentinel value of `-1` if the input string differ in number of grapheme clusters.
+*
+* @param {string} s1 - first input string
+* @param {string} s2 - second input string
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} second argument must be a string
+* @returns {integer} Hamming distance
+*
+* @example
+* var distance = hammingGraphemeClusters( 'cafe', 'café' );
+* // returns 1
+*/
+function hammingGraphemeClusters( s1, s2 ) {
+	var iend;
+	var jend;
+	var out;
+	var i;
+	var j;
+
+	if ( !isString( s1 ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', s1 ) );
+	}
+	if ( !isString( s2 ) ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be a string. Value: `%s`.', s2 ) );
+	}
+	if ( len(s1) !== len(s2) ) {
```

Comment:
You don't need to do this as a separate operation. Otherwise, this is a two-pass algorithm. This can be done in a single pass by checking whether `nextGraphemeClusterBreak` returns `-1` for either string, while the other string is still returning a nonnegative integer.

Reply:
fixed !